### PR TITLE
SEO Metatag Setting Adjustments

### LINKED ISSUE #2859: As a webmaster, I can access the advanced SEO settings so that I can hide certain pages from search
# Story

As a webmaster, I can access the advanced SEO settings so that I can hide certain landing pages from search.

Screen Shot 2021-01-13 at 3 12 48 PM

# Proposed solution

Adjust permissions for webmasters to be able to access the advanced portion of the SEO settings.

### PR DESCRIPTION
**Pull request by joewhitsitt**
Resolves #2859

Expands scope to deal with other issues...

## To Test

- Advanced robots SEO settings should be available to webmasters on article, page, person. Not sure how to test this locally to see if it actually blocks or does what it says when checked, but the configuration should be visible to webmasters.
- Person SEO Settings should now be available to webmasters, publishers, editors
- Coronavirus should have no loss in permissions for webmaster/editor as it was identical to the default config.
- Area of Study feature split should be closer to default with regard to new blocks and updates to the metatag module.
- Saving the person form/display settings on grad, admissions, immuno.grad and person_extended sites should not change anything but the metatag field being there.
- Person hometown is captured in config but only available for use on Admissions.

## Todo

core.entity_view_display.node.person.teaser

- [x] config/grad.uiowa.edu/core.entity_view_display.node.person.teaser.yml
- [x] config/admissions.uiowa.edu/core.entity_view_display.node.person.teaser.yml
- [x] config/immuno.grad.uiowa.edu/core.entity_view_display.node.person.teaser.yml
- [x] config/features/person_extended/core.entity_view_display.node.person.teaser.yml

user.role.editor

- [x] config/hr.uiowa.edu/user.role.editor.yml
- [x] config/grad.uiowa.edu/user.role.editor.yml
- [x] config/brand.uiowa.edu/user.role.editor.yml
- [x] config/features/sitenow_v2/user.role.editor.yml
- [x] config/admissions.uiowa.edu/user.role.editor.yml
- [x] config/features/scholarship/user.role.editor.yml
- [x] ~~config/coronavirus.uiowa.edu/user.role.editor.yml~~ - _identical to default, removing_
- [x] config/policy.clas.uiowa.edu/user.role.editor.yml
- [x] config/features/areas_of_study/user.role.editor.yml
- [x] config/ighn.international.uiowa.edu/user.role.editor.yml

user.role.publisher

- [x] config/hr.uiowa.edu/user.role.publisher.yml
- [x] config/brand.uiowa.edu/user.role.publisher.yml
- [x] config/admissions.uiowa.edu/user.role.publisher.yml
- [x] config/features/scholarship/user.role.publisher.yml
- [x] config/policy.clas.uiowa.edu/user.role.publisher.yml
- [x] config/features/areas_of_study/user.role.publisher.yml
- [x] config/ighn.international.uiowa.edu/user.role.publisher.yml

user.role.webmaster

- [x] config/uiowa.edu/user.role.webmaster.yml
- [x] config/features/apr/user.role.webmaster.yml
- [x] config/hr.uiowa.edu/user.role.webmaster.yml
- [x] config/grad.uiowa.edu/user.role.webmaster.yml
- [x] config/brand.uiowa.edu/user.role.webmaster.yml
- [x] config/features/sitenow_v2/user.role.webmaster.yml
- [x] config/admissions.uiowa.edu/user.role.webmaster.yml
- [x] config/features/scholarship/user.role.webmaster.yml
- [x] ~~config/coronavirus.uiowa.edu/user.role.webmaster.yml~~ - _identical to default, removing_
- [x] config/policy.clas.uiowa.edu/user.role.webmaster.yml
- [x] config/features/areas_of_study/user.role.webmaster.yml
- [x] config/ighn.international.uiowa.edu/user.role.webmaster.yml
- [x] config/presidentialsearch.uiowa.edu/user.role.webmaster.yml

core.entity_view_display.node.person.default

- [x] config/grad.uiowa.edu/core.entity_view_display.node.person.default.yml
- [x] config/admissions.uiowa.edu/core.entity_view_display.node.person.default.yml
- [x] config/immuno.grad.uiowa.edu/core.entity_view_display.node.person.default.yml
- [x] config/features/person_extended/core.entity_view_display.node.person.default.yml

core.entity_form_display.node.person.minimal.yml

- [x] config/grad.uiowa.edu/core.entity_form_display.node.person.minimal.yml
- [x] config/admissions.uiowa.edu/core.entity_form_display.node.person.minimal.yml
- [x] config/immuno.grad.uiowa.edu/core.entity_form_display.node.person.minimal.yml
- [x] config/features/person_extended/core.entity_form_display.node.person.minimal.yml

core.entity_form_display.node.person.default.yml

- [x] config/grad.uiowa.edu/core.entity_form_display.node.person.default.yml
- [x] config/admissions.uiowa.edu/core.entity_form_display.node.person.default.yml
- [x] config/immuno.grad.uiowa.edu/core.entity_form_display.node.person.default.yml
- [x] config/features/person_extended/core.entity_form_display.node.person.default.yml

### CODE REVIEW

**Comment by joewhitsitt** [comment]
Still running through a large `blt dsa` myself, but I think this is ready for review.

File: docroot/profiles/custom/sitenow/sitenow.profile [hunk 1]
```diff
@@ -378,6 +378,16 @@ function sitenow_config_split_prepare_form(EntityInterface $entity, $operation,
  * Custom node content type form defaults.
  */
 function _sitenow_node_form_defaults(&$form, $form_state) {
+  // Collapse inner SEO field groups.
```
Comment (richardbporter) [inline docroot/profiles/custom/sitenow/sitenow.profile]: Testing now. One nit though: should this be in sitenow_seo?

Comment (joewhitsitt) [reply docroot/profiles/custom/sitenow/sitenow.profile]: Yeah probably but it is unclear where the line is. I can move it when I get a chance.

**Comment by richardbporter** [comment]
Looks like it adds a meta tag to the page markup:

**Review by richardbporter — APPROVED** [review]
image

Seems to work. Some odd combinations are possible especially with the defaults of index, follow but with great power...

File: docroot/profiles/custom/sitenow/sitenow.profile (hunk 1 above)
Comment (richardbporter) [reply docroot/profiles/custom/sitenow/sitenow.profile]: Yea I agree. Like I said, a nitpick. Seems like more and more is moving aware from the profile into modules which is probably better long (long) term. :)

Comment (pyrello) [reply docroot/profiles/custom/sitenow/sitenow.profile]: This is somewhat related to #2878, which explains the rationale of why we should move this to `uiowa_core` eventually. Basically, all the modules that are enabled are dependencies of the profile, but then you have modules that might make use of core functionality, and so the profile is also technically a dependency of the module. We should be better about actually tracking those dependencies where they exist. I very much agree that this is future work though.

Comment (joewhitsitt) [reply docroot/profiles/custom/sitenow/sitenow.profile]: Sry, didn't see the latest comment. Moved to sitenow_seo cd33adb until we address the larger picture of #2878

**Pull request merged**